js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx

```diff
@@ -176,7 +176,7 @@ export function useExecutionSessionStorage(
 
   // TODO: Remove this migration logic in a few patches when we know the old namespace is likely no longer being used
   const oldDataMigrated = React.useRef(false);
-  if (oldData && !oldDataMigrated.current) {
+  if (!oldDataMigrated.current && window.localStorage.getItem(getKey(oldNamespace))) {
```

Comment:
I think we don't need the ref. If the old key is present, we save to the new key and remove the old data. All subsequent passes will be false for `getItem` on the old key, so that's our storage state right there.

Reply:
I also think this should live in an effect since it's side-effectful.

```
React.useEffect(() => {
  const oldKey = getKey(oldNamespace);
  if (!!window.localStorage.getItem(oldKey)) {
    // BEWARE! `getStorageDataForNamespace` writes to localStorage!
    const oldData = getStorageDataForNamespace(oldNamespace);
    onSave(oldData);
    window.localStorage.removeItem(oldKey);
  }
}, [oldNamespace, onSave]);
```

Reply:
AH yeah good points